tests/test_snapshot.py

```diff
@@ -431,6 +442,8 @@ async def test_snapshot_tracestats(agent, tmp_path, snapshot_ci_mode, do_referen
 
 @pytest.mark.parametrize("snapshot_removed_attrs", [{"start", "duration"}])
 async def test_removed_attributes(agent, tmp_path, snapshot_removed_attrs, do_reference_v04_http_trace):
+    resp = await agent.get("/test/session/start", params={"test_session_token": "test_case"})
```

Comment:
Is this request necessary? I worry that this will hurt the ergonomics of using the test agent if a test session token must be declared.

Reply:
The behavior is now like this:

- Request with no token => get all traces
- Request with token + token declared => get all traces directly associated with that token + all unassociated requests until the start of the next session
- Request with token + token not declared => exception (no traces found)


These are being added because the tests are submitting a request with a token, but none has ever been declared

Reply:
So I think the first two points are fine. For the third do you mean there's an exception when querying back for that token? I would prefer to have

- Request with token + token not declared (no session start) => traces for that token

That way it isn't necessary for callers to have to make a separate request.

Reply:
> Request with token + token not declared (no session start) => traces for that token

Oh, whoops, my logic and my blurb don't match:

- Request with no token => get all traces
- Request with token + token declared => get all traces directly associated with that token + all unassociated requests until the start of the next session
- Request with token + token not declared + no requests/traces for session exist directly associated => exception (no traces found)